### PR TITLE
feat: add error handling and retry mechanism in getDagMetadata API method

### PR DESCRIPTION
**Pull request by pedroferreira1**

### CODE REVIEW

File: src/api/metadataApi.ts [hunk 1]
```diff
@@ -6,11 +6,48 @@
...
+          // throw Error and the catch will handle it
+          throw Error;
+        }
+      } catch (e) {
+        if (e.response && e.response.status === 404) {
+          // No need to do anything, the metadata for this token was not found
+          // There is no error here, we just return null
+          return null;
+        } else {
+          // Error downloading metadata, then we should wait a few seconds and retry if still didn't reached retry limit
```
Comment (andreabadesso) [inline src/api/metadataApi.ts:41]: I suggest changing the position of this comment to the line just above

`await helpers.sleep(retryInterval);`

And add another comment explaining that when `retries === 0` the method will propagate the error

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/api/metadataApi.ts (hunk 1 above)
Comment (pedroferreira1) [reply src/api/metadataApi.ts]: Improved the comments here: https://github.com/HathorNetwork/hathor-wallet-lib/pull/266/commits/0085f4c46108282fde98a7bcb9e536d48d0a8e2d

What do you think?

**Review by andreabadesso — APPROVED** [review]

File: src/api/metadataApi.ts [hunk 2]
```diff
@@ -6,11 +6,50 @@
...
+    while (retries >= 0) {
+      const axios = await explorerServiceAxios(network);
+      try {
+        const response = await axios.get(`metadata/dag`, { params: { id }});
+        if (response.data) {
+          return response.data;
+        } else {
+          // Error downloading metadata
+          // throw Error and the catch will handle it
+          throw Error;
```
Comment (msbrogli) [inline src/api/metadataApi.ts:33]: Should we customize the error? We can also add a message to it but I feel a customized error is better.

**Review by msbrogli — APPROVED** [review]

File: src/api/metadataApi.ts (hunk 2 above)
Comment (pedroferreira1) [reply src/api/metadataApi.ts]: Done https://github.com/HathorNetwork/hathor-wallet-lib/pull/266/commits/1c957a38e489e17807245a45df8f97105841e99d

**Pull request merged**